cli/lib/kontena_cli.rb

```diff
@@ -1,32 +1,41 @@
 module Kontena
-  # Run a kontena command like it was launched from the command line.
-  #
-  # @example
-  #   Kontena.run("grid list --help")
+  # Run a kontena command like it was launched from the command line. Re-raises any exceptions,
+  # except a SystemExit with status 0, which is considered a success.
   #
   # @param [String,Array<String>] command_line
-  # @return [Fixnum] exit_code
-  def self.run(*cmdline, returning: :status)
+  # @return command result or nil
+  def self.run(*cmdline)
     if cmdline.first.kind_of?(Array)
       command = cmdline.first
     elsif cmdline.size == 1 && cmdline.first.include?(' ')
       command = cmdline.first.shellsplit
     else
       command = cmdline
     end
-    ENV["DEBUG"] && puts("Running Kontena.run(#{command.inspect}, returning: #{returning}")
+    ENV["DEBUG"] && puts("Running Kontena.run(#{command.inspect}")
     result = Kontena::MainCommand.new(File.basename(__FILE__)).run(command)
     ENV["DEBUG"] && puts("Command completed, result: #{result.inspect} status: 0")
-    return 0 if returning == :status
-    return result if returning == :result
+    result
   rescue SystemExit => ex
-    ENV["DEBUG"] && $stderr.puts("Command completed with failure, result: #{result.inspect} status: #{ex.status}")
-    returning == :status ? $!.status : nil
+    ENV["DEBUG"] && $stderr.puts("Command caused SystemExit, result: #{result.inspect} status: #{ex.status}")
+    return true if ex.status.zero?
+    raise ex
   rescue => ex
-    ENV["DEBUG"] && $stderr.puts("Command raised #{ex} with message: #{ex.message}\n#{ex.backtrace.join("\n  ")}")
-    returning == :status ? 1 : nil
+    ENV["DEBUG"] && $stderr.puts("Command raised #{ex.class.name} with message: #{ex.message}\n#{ex.backtrace.join("\n  ")}")
+    raise ex
   end
 
+  # Run a kontena command and return true if the command did not raise or exit with a non-zero exit code. Raises nothing.
+  # @param [String,Array<String>] command_line
+  # @return [TrueClass,FalseClass] success
+  def self.run?(*cmdline)
```

Comment:
Just nitpicking on naming now, but having `run?` as a variant of `run` that doesn't raise exceptions, but still has side-effects, doesn't seem right. Shouldn't it be `run!` that raises, and then a `run` wrapper that rescues the `SystemExit` to return `false`? Most callers would then call `run!`.

Although, if I understand the callers correctly, then it might also work to just have a single `run` method that re-raised `SystemExit` as a `StandardError`?